bin/install.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+set -eu
+
+# Ensure PHP is installed
+if ! command -v php > /dev/null 2>&1; then
+  echo "PHP is not found, installation aborted!"
+  exit 1
+fi
+
+# Ensure Composer is installed/updated
+if ! command -v composer > /dev/null 2>&1; then
+  echo "*** Installing Composer ***"
+  php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
+
+  if [ \
+      "$(curl -S https://composer.github.io/installer.sig)" = \
+      "$(php -r \"echo hash_file\('SHA384', 'composer-setup.php'\);\")" \
+  ]; then
+      php composer-setup.php --quiet --filename=composer
+      sudo mv composer /usr/local/bin/composer
+      rm composer-setup.php
+  else
+      >&2 echo 'ERROR: Invalid composer installer signature'
+      rm composer-setup.php
+      exit 1
+  fi
+fi
+
+echo "*** Updating shell profiles ***"
+# Add Composer's Global Bin to ~/.profile path
+if [[ ! -f "$HOME/.profile" ]]; then
+  touch "$HOME/.profile"
+fi
+if grep -q -F "export COMPOSER_HOME=$HOME/.composer" "$HOME/.profile"; then
+  echo "export COMPOSER_HOME=$HOME/.composer" >> "$HOME/.profile"
+fi
+if grep -q -F "export PATH=$PATH:\$COMPOSER_HOME/vendor/bin" "$HOME/.profile"; then
+  echo "export PATH=$PATH:\$COMPOSER_HOME/vendor/bin" >> "$HOME/.profile"
+fi
+
+# Source the .profile to pick up changes
+# shellcheck source=/dev/null
+source "$HOME/.profile"
```

Comment:
This _could_ have unintended side-effects, depending on what the user has (been told by other programs to) put in their `.profile`